tests/Google/Ads/GoogleAds/Lib/OAuth2TokenBuilderTest.php

```diff
@@ -65,4 +70,117 @@ public function testBuildFailsWhenMissingRequiredValuesForInstAppOrWebFlow()
             ->withClientSecret('ABcXyZ-123abc')
             ->build();
     }
+
+    /**
+     * @covers \Google\Ads\GoogleAds\Lib\OAuth2TokenBuilder::from
+     */
+    public function testBuildFromWithServiceAccountFlow()
+    {
+        $valueMap = [
+            [
+                'jsonKeyFilePath',
+                'OAUTH2',
+                $this->jsonKeyFilePath
+            ], [
+                'scopes',
+                'OAUTH2',
+                'https://www.googleapis.com/auth/adwords'
+            ],
+        ];
+        $configurationMock = $this->getMockBuilder(Configuration::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $configurationMock->expects($this->any())
+            ->method('getConfiguration')
+            ->will($this->returnValueMap($valueMap));
+        $tokenFetcher = $this->oAuth2TokenBuilder
+            ->from($configurationMock)
+            ->build();
+        $this->assertInstanceOf(ServiceAccountCredentials::class, $tokenFetcher);
+    }
+
+    /**
+     * @covers \Google\Ads\GoogleAds\Lib\OAuth2TokenBuilder::from
+     */
+    public function testBuildFromWithServiceAccountFlowUsingImpersonation()
+    {
+        $valueMap = [
+            [
+                'jsonKeyFilePath',
+                'OAUTH2',
+                $this->jsonKeyFilePath
+            ], [
+                'scopes',
+                'OAUTH2',
+                'https://www.googleapis.com/auth/adwords'
+            ], [
+                'impersonatedEmail',
+                'OAUTH2',
+                'gaapi@gmail.com'
```

Comment:
This is not an actual email, is it? I would hate for someone to get spammed if a bot or a copy-paste person reads this. I know I'm being cautious, but if possible, something completely non-existent would be nice.

Reply:
Good point but I really don't know too. We may try creating a new email using that one but it'd be a bit overkill in this case, so I changed to really long one that shouldn't be used by anyone instead. :)